supergloo/docs/mesh/tutorials-3-prometheus-metrics.md

```diff
@@ -0,0 +1,151 @@
+
+```yaml
+kubectl get configmap -n prometheus-test -o yaml | grep istio
+      - job_name: supergloo-istio-supergloo-system.istio-envoy-stats
+      - job_name: supergloo-istio-supergloo-system.istio-galley
+            - istio-system
+          regex: istio-galley;http-monitoring
+      - job_name: supergloo-istio-supergloo-system.istio-istio-mesh
+            - istio-system
+          regex: istio-telemetry;prometheus
+      - job_name: supergloo-istio-supergloo-system.istio-istio-policy
+            - istio-system
+          regex: istio-policy;http-monitoring
+      - job_name: supergloo-istio-supergloo-system.istio-istio-telemetry
+            - istio-system
+          regex: istio-telemetry;http-monitoring
+      - job_name: supergloo-istio-supergloo-system.istio-pilot
+            - istio-system
+          regex: istio-pilot;http-monitoring
+```
+
+We can see the configuration that this applied to our Mesh CRD by running:
+
+{{< highlight yaml "hl_lines=16-19" >}}
+kubectl get mesh -n supergloo-system istio -o yaml
+
+apiVersion: supergloo.solo.io/v1
+kind: Mesh
+metadata:
+  creationTimestamp: 2019-03-28T18:44:46Z
+  generation: 1
+  name: istio
+  namespace: supergloo-system
+  resourceVersion: "178284"
+  selfLink: /apis/supergloo.solo.io/v1/namespaces/supergloo-system/meshes/istio
+  uid: 8f57f47e-5189-11e9-9c12-b0cb59a58200
+spec:
+  istio:
+    installationNamespace: istio-system
+  monitoringConfig:
+    prometheusConfigmaps:
+    - name: prometheus-server
+      namespace: prometheus-test
+  mtlsConfig:
+    mtlsEnabled: true
+status:
+  reported_by: istio-config-reporter
+  state: 1
+{{< /highlight >}}
+
+Notice how the `monitoringConfig` now contains an entry for our `prometheus-server` configmap.
+
+Let's take a look at the metrics that our Prometheus instance should have started collecting for our mesh.
+
+Open up a port-forward to reach the Prometheus UI from our local machine:
+
+```yaml
+kubectl port-forward -n prometheus-test deployment/prometheus-server 9090
+```
+
+Now direct your browser to http://localhost:9090/
+
+You should see the Prometheus Graph page show up:
+
+![Prometheus Landing Page](../../img/prometheus-landing-page.png "Prometheus Landing Page")
+
+Let's enter a query to see some stats from Istio. We'll try `istio_requests_total`:
+
+![Prometheus Initial Query](../../img/prometheus-initial-query.png "Prometheus Initial Query")
+
+Let's try creating some metrics by sending traffic to some of our Bookinfo pods. Open the port-forward to 
+reach the productpage:
+
+```bash
+kubectl port-forward -n default deployment/productpage-v1 9080
+```
+
+Open your browser to http://localhost:9080/productpage. Refresh the page a few times - 
+this will cause the product page to send requests to the reviews and ratings services.
+
+Now let's check back in Prometheus and try the query `istio_requests_total{destination_app="reviews"}` 
+(note that it might take up to 30 seconds before new metrics are scraped by Prometheus):
```

Comment:
It also might be good to suggest changing the time window down from the default 1hr to like 5m to see the graph a bit more clearly

Reply:
i'm not that familiar with PromQL to do such a query. i tried `istio_requests_total[5m]` and got **Error executing query: invalid expression type "range vector" for range query, must be Scalar or instant Vector**

do you have an example query i can try?

Reply:
Maybe just make a note that they should be aware of time time window?